components/raftstore/src/store/fsm/peer.rs

```diff
@@ -859,6 +873,48 @@ where
         self.register_raft_base_tick();
     }
 
+    fn finish_unsafe_recovery_wait_apply(&mut self) {
+        if self
+            .fsm
+            .unsafe_recovery_wait_apply_counter
+            .as_ref()
+            .unwrap()
+            .fetch_sub(1, Ordering::Relaxed)
+            == 1
+        {
+            let stats = StoreStats::default();
```

Comment:
The default `StoreStats` does not set the `store_id` field, may cause PD confuse about where the store heartbeat comes from.

Reply:
Done

Reply:
I still worry about a default `StoreStats` that many fields like `bytes_written` are set to 0 will cause PD to make some wrong schedule decision.

Reply:
How about introducing a new `StoreMsg` to let the store fsm (reuse `store_heartbeat_pd`) send a `StoreHeartbeat` (with `send_detailed_report: true`) to the PD worker immediately.

Reply:
Currently, it has been taken care of, see the PD [code here](https://github.com/tikv/pd/blob/master/server/grpc_service.go#L598). The best way to address that concern, while not bringing in other side effects, is to use a separate RPC instead of the heartbeat. This can be done as a optimization in the future? What do you think? @Connor1996

Reply:
Yes, let's do it then if it's really an issue.